src/cache/cache_las.c

```diff
@@ -1269,11 +1269,11 @@ __wt_find_lookaside_upd(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDAT
         break;
     case BTREE_COL_FIX:
     case BTREE_COL_VAR:
-        WT_RET(__wt_buf_grow(session, cbt->tmp, WT_INTPACK64_MAXSIZE));
-        p = cbt->tmp->mem;
+        WT_RET(__wt_buf_grow(session, cbt->las, WT_INTPACK64_MAXSIZE));
+        p = cbt->las->mem;
         WT_RET(__wt_vpack_uint(&p, 0, cbt->recno));
-        cbt->tmp->size = WT_PTRDIFF(p, cbt->tmp->mem);
-        key = cbt->tmp;
+        cbt->las->size = WT_PTRDIFF(p, cbt->las->mem);
+        key = cbt->las;
```

Comment:
There are a lot of cursors in the system (thousands and thousands), we don't want to grow that structure. We only need a small buffer here. I think something like this would be sufficient:
```
*** ../cache/cache_las.c        Wed Oct  2 01:30:49 2019
--- -   Wed Oct  2 01:36:35 2019
***************  
*** 1233,1246 ****
      WT_DECL_ITEM(las_key);
      WT_DECL_ITEM(las_value);
      WT_DECL_RET;
!     WT_ITEM *key;
      WT_TXN *txn; 
      WT_UPDATE *upd, *list[WT_MODIFY_ARRAY_SIZE], **listp;
      wt_timestamp_t durable_timestamp, _durable_timestamp, las_timestamp, _las_timestamp;
      size_t allocated_bytes, incr;
      uint64_t las_txnid, _las_txnid;
      uint32_t las_btree_id, session_flags;
!     uint8_t prepare_state, _prepare_state, *p, upd_type;
      u_int i;
      int cmp;
      bool sweep_locked;
--- 1233,1246 ----
      WT_DECL_ITEM(las_key);
      WT_DECL_ITEM(las_value);
      WT_DECL_RET;
!     WT_ITEM key;
      WT_TXN *txn;
      WT_UPDATE *upd, *list[WT_MODIFY_ARRAY_SIZE], **listp;
      wt_timestamp_t durable_timestamp, _durable_timestamp, las_timestamp, _las_timestamp;
      size_t allocated_bytes, incr;
      uint64_t las_txnid, _las_txnid;
      uint32_t las_btree_id, session_flags;
!     uint8_t prepare_state, _prepare_state, *p, recno_key[WT_INTPACK64_MAXSIZE], upd_type;
      u_int i;
      int cmp;
      bool sweep_locked;
***************
*** 1269,1279 ****
          break;
      case BTREE_COL_FIX:
      case BTREE_COL_VAR:
!         WT_RET(__wt_buf_grow(session, cbt->las, WT_INTPACK64_MAXSIZE));
!         p = cbt->las->mem;
          WT_RET(__wt_vpack_uint(&p, 0, cbt->recno));
!         cbt->las->size = WT_PTRDIFF(p, cbt->las->mem);
!         key = cbt->las;
      }

      /* Allocate buffers for the lookaside key/value. */
--- 1269,1278 ----
          break;
      case BTREE_COL_FIX:
      case BTREE_COL_VAR:
!         p = recno_key;
          WT_RET(__wt_vpack_uint(&p, 0, cbt->recno));
!         key.data = recno_key;
!         key.size = WT_PTRDIFF(p, recno_key);
      }

      /* Allocate buffers for the lookaside key/value. */
```
That's untested, but I think it's close.

Reply:
Thanks for the suggestion, yes it is a very good improvement. I changed the code to use the local variable as you suggested.